### PR TITLE
txn: force pessimistic lock

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

###  What have you changed?
Force pessimistic lock.

### CODE REVIEW

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -267,7 +274,7 @@ impl<S: Snapshot> MvccTxn<S> {
                     .acquire_pessimistic_lock
                     .inc();
             }
-            return Ok(());
+            return res;
         }
 
         if let Some((commit_ts, write)) = self.reader.seek_write(&key, TimeStamp::max())? {
```
Comment (Little-Wallace) [inline src/storage/mvcc/txn.rs]: We have read data here, why call `force_get` in `pessimistic_lock`?

File: src/storage/mvcc/txn.rs [hunk 2]
```diff
@@ -258,6 +260,11 @@ impl<S: Snapshot> MvccTxn<S> {
                 }
                 .into());
             }
+            let res = if force {
+                Ok(Some(self.reader.force_get(&key)?))
```
Comment (Little-Wallace) [inline src/storage/mvcc/txn.rs]: Could we read this value in 
```
if let Some((commit_ts, write)) = self.reader.seek_write(&key, TimeStamp::max())? {
```

File: src/storage/mvcc/txn.rs [hunk 3]
```diff
@@ -320,10 +327,14 @@ impl<S: Snapshot> MvccTxn<S> {
             self.check_data_constraint(should_not_exist, &write, commit_ts, &key)?;
         }
 
+        let res = if force {
```
Comment (Little-Wallace) [inline src/storage/mvcc/txn.rs]: Could we use result in last step?
```
if let Some((commit_ts, write)) = self.reader.seek_write(&key, TimeStamp::max())? {
```

Comment (youjiali1995) [reply src/storage/mvcc/txn.rs]: I added a quick path for it.

**Pull request closed without merging**